### ISSUE TITLE
ES acts strange for queries based on numbers

### ISSUE THREAD

**Issue by mariacarmina** [oceanprotocol/aquarius#584]
On Discord, a user had problem reproducing some queries that I have listed below and @calina-c provided some support on it:

- https://aquarius.oceanprotocol.com/api/v1/aquarius/assets/query%7B%22query%22:%7B%22query_string%22:%7B%22query%22:%2244.43697584546595%22%7D%7D%7D;

- https://aquarius.oceanprotocol.com/api/v1/aquarius/assets/query%7B%22query%22:%7B%22query_string%22:%7B%22fields%22:[%22service.attributes.additionalInformation.workExample%22],%22query%22:%2244.43*%22%7D%7D%7D

- https://aquarius.oceanprotocol.com/api/v1/aquarius/assets/query%7B%22query%22:%22query_string%22:%7B%22query": "(service.attributes.additionalInformation.description:"Weather information of UK including temperature and humidity") -isInPurgatory:true"}},"sort": {created: 'desc'}}

> Note that at that time the path  was different
https://aquarius.rinkeby.oceanprotocol.com/api/v1/aquarius/assets/ddo/query

**JSON payloads**
`const query = {
                "query": {
                    "query_string": {
                          "fields"  : ["service.attributes.additionalInformation.workExample"],
                            "query": "45.75"
                    }
                }
            };`

       ` const query5 = {
                "query": {
                    "query_string": {
                        "query": "(service.attributes.additionalInformation.description:'Weather information of UK including temperature and humidity') -isInPurgatory:true",
                    }
                },
                sort: { created: 'desc' }
            };`

       ` const query3 = {
                "query": {
                    "query_string": {
                        "query": "(service.attributes.additionalInformation.workExample:'45.75') -isInPurgatory:true",
                    }
                },
                sort: { created: 'desc' }
            };`

        `const query3a = {
                "query": {
                    "query_string": {
                        "query": "(service.attributes.additionalInformation.workExample:'45.75') -isInPurgatory:true",
                    }
                },
                sort: { created: 'desc' }
            };`

       `const query2 = {
                "query": {
                    "query_string": {
                        "query": "(service.attributes.additionalInformation.workExample:'45.75347943774362') -isInPurgatory:true",
                    }
                }
            };`

**Logs of a successful query sample**:

`01-29 20:07:02.965: I/NODEJS-MOBILE(14336): ===========================query==============================
01-29 20:07:02.965: I/NODEJS-MOBILE(14336): { page: 1,
01-29 20:07:02.965: I/NODEJS-MOBILE(14336):   offset: 100,
01-29 20:07:02.965: I/NODEJS-MOBILE(14336):   query: 
01-29 20:07:02.965: I/NODEJS-MOBILE(14336):    { nativeSearch: 1,
01-29 20:07:02.965: I/NODEJS-MOBILE(14336):      query_string: 
01-29 20:07:02.965: I/NODEJS-MOBILE(14336):       { query: '(service.attributes.additionalInformation.workExample:\'44.43\'*) -isInPurgatory:true',
01-29 20:07:02.965: I/NODEJS-MOBILE(14336):         analyze_wildcard: true } },
01-29 20:07:02.965: I/NODEJS-MOBILE(14336):   sort: { created: -1 } }
01-29 20:07:02.966: I/NODEJS-MOBILE(14336): ===========================emptyString + this.url + apiPath + /query==============================
01-29 20:07:02.966: I/NODEJS-MOBILE(14336): https://aquarius.rinkeby.oceanprotocol.com/api/v1/aquarius/assets/ddo/query
01-29 20:07:02.966: I/NODEJS-MOBILE(14336): {"page":1,"offset":100,"query":{"nativeSearch":1,"query_string":{"query":"(service.attributes.additionalInformation.workExample:'44.43'*) -isInPurgatory:true","analyze_wildcard":true}},"sort":{"created":-1}}
01-29 20:07:03.755: I/NODEJS-MOBILE(14336): ===========================apsCloseToCurrentGpsLocationArray==============================
01-29 20:07:03.764: I/NODEJS-MOBILE(14336): { results: 
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):    [ DDO {
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        '@context': 'https://w3id.org/did/v1',
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        id: 'did:op:8e534A7eB121c1c926A5f91A0A1Fa21ab5fE1EE4',
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        publicKey: [Array],
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        authentication: [Array],
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        service: [Array],
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        dataToken: '0x8e534A7eB121c1c926A5f91A0A1Fa21ab5fE1EE4',
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        created: '2021-01-24T01:03:38Z',
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        proof: [Object],
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        dataTokenInfo: [Object],
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        updated: '2021-01-24T01:03:38Z',
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        accessWhiteList: [],
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        price: [Object],
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        isInPurgatory: 'false' },
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):      DDO {
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        '@context': 'https://w3id.org/did/v1',
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        id: 'did:op:8d11Ce389C7B906Ece56a2EdcB0e10bf80D9FC20',
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        publicKey: [Array],
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        authentication: [Array],
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        service: [Array],
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        dataToken: '0x8d11Ce389C7B906Ece56a2EdcB0e10bf80D9FC20',
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        created: '2021-01-24T01:03:08Z',
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        proof: [Object],
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        dataTokenInfo: [Object],
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        updated: '2021-01-24T01:03:08Z',
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        accessWhiteList: [],
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        price: [Object],
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        isInPurgatory: 'false' },
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):      DDO {
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        '@context': 'https://w3id.org/did/v1',
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        id: 'did:op:3d56568D9a01Ed73088d8dBa2563631Be31e3AD5',
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        publicKey: [Array],
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        authentication: [Array],
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        service: [Array],
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        dataToken: '0x3d56568D9a01Ed73088d8dBa2563631Be31e3AD5',
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        created: '2021-01-23T22:13:05Z',
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        proof: [Object],
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        dataTokenInfo: [Object],
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        updated: '2021-01-23T22:13:05Z',
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        accessWhiteList: [],
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        price: [Object],
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        isInPurgatory: 'false' },
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):      DDO {
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        '@context': 'https://w3id.org/did/v1',
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        id: 'did:op:937ad6FD1D8870438a2196f24cC4b7E1968Db8A2',
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        publicKey: [Array],
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        authentication: [Array],
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        service: [Array],
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        dataToken: '0x937ad6FD1D8870438a2196f24cC4b7E1968Db8A2',
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        created: '2021-01-23T22:10:05Z',
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        proof: [Object],
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        dataTokenInfo: [Object],
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        updated: '2021-01-23T22:10:05Z',
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        accessWhiteList: [],
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        price: [Object],
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        isInPurgatory: 'false' },
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):      DDO {
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        '@context': 'https://w3id.org/did/v1',
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        id: 'did:op:8bDB1B8
01-29 20:07:03.764: I/NODEJS-MOBILE(14336): 39162D142435aE4ab578Ef3BA5D1dBb45',
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        publicKey: [Array],
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        authentication: [Array],
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        service: [Array],
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        dataToken: '0x8bDB1B839162D142435aE4ab578Ef3BA5D1dBb45',
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        created: '2021-01-23T22:08:05Z',
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        proof: [Object],
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        dataTokenInfo: [Object],
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        updated: '2021-01-23T22:08:05Z',
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        accessWhiteList: [],
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        price: [Object],
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        isInPurgatory: 'false' },
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):      DDO {
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        '@context': 'https://w3id.org/did/v1',
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        id: 'did:op:8002d7DbD5fCa96314739Ee5194A683Fa7630eBC',
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        publicKey: [Array],
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        authentication: [Array],
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        service: [Array],
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        dataToken: '0x8002d7DbD5fCa96314739Ee5194A683Fa7630eBC',
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        created: '2021-01-21T18:06:22Z',
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        proof: [Object],
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        dataTokenInfo: [Object],
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        updated: '2021-01-21T18:06:22Z',
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        accessWhiteList: [],
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        price: [Object],
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        isInPurgatory: 'false' },
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):      DDO {
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        '@context': 'https://w3id.org/did/v1',
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        id: 'did:op:0281defB5Fa179aA5b727c759245aB50D1b3d452',
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        publicKey: [Array],
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        authentication: [Array],
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        service: [Array],
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        dataToken: '0x0281defB5Fa179aA5b727c759245aB50D1b3d452',
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        created: '2021-01-21T18:03:37Z',
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        proof: [Object],
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        dataTokenInfo: [Object],
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        updated: '2021-01-21T18:03:37Z',
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        accessWhiteList: [],
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        price: [Object],
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        isInPurgatory: 'false' },
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):      DDO {
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        '@context': 'https://w3id.org/did/v1',
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        id: 'did:op:20F617773F853ae366174259Bb88e85f7eC2f1b2',
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        publicKey: [Array],
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        authentication: [Array],
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        service: [Array],
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        dataToken: '0x20F617773F853ae366174259Bb88e85f7eC2f1b2',
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        created: '2021-01-21T18:03:07Z',
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        proof: [Object],
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        dataTokenInfo: [Object],
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        updated: '2021-01-21T18:03:07Z',
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        accessWhiteList: [],
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        price: [Object],
01-29 20:07:03.764: I/NODEJS-MOBILE(14336):        isInPurgatory: 'false' },`

**NOTE**: The user used RESer Firefox extension to run POST executions.

**Comment by mihaisc** [comment]
What are the first 3 links? we don't do anything with the query string, the query should be in the body. Am i misunderstanding something?

**Comment by mariacarmina** [comment]
As far as I understood, the first 3 links are some queries attempts from the user which do not work from a specific browser extension. @calina-c had more context regarding this.

**Comment by mihaisc** [comment]
for example this query is ok
```
{
    "query": {
        "query_string": {
            "query": "(service.attributes.additionalInformation.workExample:\"[42.35427883472187, -71.17704939789566]\") -isInPurgatory:true"
        }
    },
    "sort": {
        "created": "desc"
    }
}
```
`workExample` is not a number is just a string.

**Comment by mihaisc** [comment]
As per documentation https://deploy-preview-783--docs-oceanprotocol.netlify.app/references/aquarius/#apiv1aquariusassetsquery , query accepts post, not get. Not sure where the issue is. We need more details on what the issue is.

**Comment by adonnini** [comment]
I have run the queries listed above both from my app, and interactively using POST not GET. The result is the same, 404.

Previously 1/29/21), as per the sample output above, the same queries ran successfully returning non-empty results.

When I run query interactively, I use the RETSer extension in Firefox. Here is a sample command line which returns 404

Clearly, I am doing something wrong because when I run

POST https://aquarius.oceanprotocol.com/api/v1/aquarius/assets/query{"query": {"query_string": {"query": "(service.attributes.additionalInformation.workExample:\"[42.35427883472187, -71.17704939789566]\") -isInPurgatory:true"}},sort": {"created": "desc"}}
using RESTer
the result is 404

When I run queries programmatically here is a sample call

	const q2 ={
	query:{
    		"query_string" : {
        			    "query" :"45.753479"+"*",
        			    "fields"  : ["service.attributes.additionalInformation.workExample"]
    	}
 	}
	};
	apsCloseToCurrentGpsLocation = await metadataCache.queryMetadata(q2);	

I did not change MetadataCache except for adding the following line

const apiPathForQueryMetadata = '/api/v1/aquarius/assets'

and using apiPathForQueryMetadata in queryMetadata instead of apiPath

I realize that I must be doing something wrong, especially since you have a query structured like the ones I have which produces results. I don't know what it is. I am puzzled by the fact that queries that used to run (as per the sample output) do not any longer.

I will try to run the query you were able to run successfully from my app.

Thanks for your help.

**Comment by adonnini** [comment]
May I ask why in the test query you ran you preceded the " with a \ in
\"[42.35427883472187, -71.17704939789566]\"
?
Thanks

**Comment by adonnini** [comment]
Update,
logs in my pp report that the following query was run by queryMetadata : 
===========================emptyString + this.url + apiPathForQueryMetadata + /query==============================
https://aquarius.oceanprotocol.com//api/v1/aquarius/assets/query
{"query":{"query_string":{"query":"(service.attributes.additionalInformation.workExample:\"[42.35427883472187, -71.17704939789566]\") -isInPurgatory:true"}},"sort":{"created":"desc"}}

The query reported no results. 

I am sorry to be taking up time with this

**Comment by mihaisc** [comment]
Because it is most probably indexed as a string by default.

**Comment by adonnini** [comment]
I am sorry I am not sure I understand. What is the "it" you refer to, workExample? How would I determine whether it is indexed as a string by default? How did you run it so that it worked for you while I could not make it work? I still do not understand why queries like the ones I am running sued to work and don' any longer.

What does it mean in terms of steps I should take to resolve my problem?

Sorry for all these questions.

Thanks

**Comment by adonnini** [comment]
I reread
https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-query-string-query.html
and
https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-query-string-query.html#query-string-syntax
If I understand the docs correctly I do not have to specify a field value

I ran the following query via Postman
POST https://aquarius.oceanprotocol.com/api/v1/aquarius/assets/query{"query": {"query_string": {"query": "RsaSignatureAuthentication2018"}},sort": {"created": "desc"}}

Unfortunately, the result was:

<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN">
<title>404 Not Found</title>
<h1>Not Found</h1>
<p>The requested URL was not found on the server. If you entered the URL manually please check your spelling and try again.</p>

Is there something wrong with the base URL?

Thanks

**Comment by adonnini** [comment]
Setting a fields parameter in the above query to
"fields" : ["authentication.type"]
made no difference

**Comment by adonnini** [comment]
I don't know if this is relevant, when I run a retrieval like this
https://aquarius.oceanprotocol.com/api/v1/aquarius/assets/ddo/did:op:95B846f33a2D9B06d2E808c26990Ac66B16F9aA5
The result is displayed as a single string. Until about a couple of weeks ago, it was displayed as a JSON payload with tabs giveing you the option to see it raw, etc., etc.

**Comment by mihaisc** [comment]
Like i said in a previous comment the url you are using is incorrect so it will not work . A quick export from postman 
```
curl --location --request POST 'https://aquarius.oceanprotocol.com/api/v1/aquarius/assets/query' \
--header 'Content-Type: application/json' \
--data-raw '{
    "query": {
        "query_string": {
            "query": "(service.attributes.additionalInformation.workExample:\"[42.35427883472187, -71.17704939789566]\") -isInPurgatory:true"
        }
    },
    "sort": {
        "created": "desc"
    }
}'
```
In postman the query should be written in Body / raw  with json format
![image](https://user-images.githubusercontent.com/26000280/135411195-87231391-08d8-4372-b790-6abbe4aa1601.png)

**Comment by adonnini** [comment]
Thanks. I was/am using Postman to determine whether a query I am using in my app is valid and returns non-empty results.
My goal is to make the query work when executed from the app.
I am sorry. I looked through the messages above and could not find the comment about the URL being wrong. Is it this one?

"As per documentation https://deploy-preview-783--docs-oceanprotocol.netlify.app/references/aquarius/#apiv1aquariusassetsquery , query accepts post, not get. Not sure where the issue is. We need more details on what the issue is."

What is wrong with the URL?

If my understanding is correct, when a query it is submitted programmatically, it is built by this line in MetadataCache.ts (in ocean.js)

post(`${this.url}${apiPath}/query`, JSON.stringify(query))

where 

this.url is https://aquarius.oceanprotocol.com
apiPath is /api/v1/aquarius/assets/
and
JSON.stringify(query) is the srting version
of a query, e.g.
{"query": {
        "query_string": {
            "query": "(service.attributes.additionalInformation.workExample:\"[42.35427883472187, -71.17704939789566]\") -isInPurgatory:true"
        }
    },
    "sort": {
        "created": "desc"
    }
}

Thanks

**Comment by adonnini** [comment]
FYI, I just ran all the queries I have in the app using Postman (correctly). They all work as expected.

**Comment by adonnini** [comment]
queryMetadata in ocean.js uses a WebServiceConnector to post the query.

queryMetadata in Aquarius.ts uses Axios

Could this be a reason for the failure of the query?

**Comment by adonnini** [comment]
Hi @mihaisc Sorry to bug you. I can imagine you are busy with higher priority tasks. I am really stuck. When do you think you will be able to send me feedback on my latest additions to this issue(please see above)?
Thanks,
Alex Donnini

**Comment by alexcos20** [comment]
what version of ocean.js are you using?

**Comment by adonnini** [comment]
17.1
If it helps, I do not believe MetadataCache.ts has changed since then, but I could be wrong

**Comment by alexcos20** [comment]
yes, it does   https://github.com/oceanprotocol/ocean.js/commit/3bea2aaf9082bf1b61219f3d7960c14cec04fa1e#diff-3da6030da4b8f36a5982b8d19d8a1a11fd6a73f06c6fda80c10956b1f84a78ffR91

until 0.18.0, we were using /api/v1/aquarius/assets/ddo/query.   now we are using /api/v1/aquarius/assets/query

pls bump to 0.18.0

**Comment by adonnini** [comment]
I am using /api/v1/aquarius/assets/query ( added a varaible to my copy of MetadatCache.ts called

apiPathForQueryMetadata whose value I set to '/api/v1/aquarius/assets'

If I weren't the attempt to run query would fail.

In my case it does not fail. It returns no results

I will bump up tp 0.18.0

**Comment by adonnini** [comment]
I am using /api/v1/aquarius/assets/query ( added a varaible to my copy 
of MetadatCache.ts called

apiPathForQueryMetadata whose value I set to '/api/v1/aquarius/assets'

If I weren't the attempt to run query would fail.

In my case it does not fail. It returns no results

I will bump up tp 0.18.0


Alex Coseru wrote:

**Comment by adonnini** [comment]
While upgrading to 0.18, I ran into a problem which I need to resolve in order to be able to complete the upgrade.

It might make sense to close this issue.

Once I am able to complete the upgrade, I will run the app (and query) again. If it doesn't work, I will open another issue.

Thanks for your help

**Comment by adonnini** [comment]
This is incorrect.I used the right api all along. It's possible that as suggested upgrading to 0.18 would resolve the problem, not because of the api but by due to an upgrade of web3.I suggested that the issue be closed because I'm having a problem upgrading to 0.18 not related to ocean protocol. Rather than wait to resolve my problem I thought it would be better to close the issue.I will open one again if the upgrade to 0.18 does not resolve the problem. Thank you On 6 Oct 2021 09:53, Matthias Kretschmann ***@***.***> wrote:
as far as I see the original issue was caused by using an outdated API path so closing this

**Comment by adonnini** [comment]
This conclusion is incorrect.
I used the right api all along. 
It's possible that as suggested upgrading to 0.18 would resolve the problem, not because of the api but by due to an upgrade of web3.
I suggested that the issue be closed because I'm having a problem upgrading to 0.18 not related to ocean protocol. 
Rather than wait to resolve my problem I thought it would be better to close the issue.
I will open one again if the upgrade to 0.18 does not resolve the problem. 
Thank you
